src/hooks/useCourtContracts.js

```diff
@@ -386,27 +419,34 @@ export function useTotalActiveBalancePolling(termId) {
 
   const timeoutId = useRef(null)
 
-  const fetchTotalActiveBalance = useCallback(() => {
-    timeoutId.current = setTimeout(() => {
-      return jurorRegistryContract
-        .totalActiveBalanceAt(termId)
-        .then(balance => {
-          setTotalActiveBalance(balance)
-          clearTimeout(timeoutId.current)
-          fetchTotalActiveBalance()
-        })
-        .catch(err => {
-          console.log(`Error fetching balance: ${err} retrying...`)
-          fetchTotalActiveBalance()
-        })
-    }, 1000)
-  }, [jurorRegistryContract, termId])
-
   useEffect(() => {
+    let cancelled = false
+
+    const fetchTotalActiveBalance = () => {
+      timeoutId.current = setTimeout(() => {
+        return jurorRegistryContract
+          .totalActiveBalanceAt(termId)
+          .then(balance => {
+            if (!cancelled) {
+              setTotalActiveBalance(balance)
+              clearTimeout(timeoutId.current)
+              fetchTotalActiveBalance()
+            }
+          })
+          .catch(err => {
+            console.error(`Error fetching balance: ${err} retrying...`)
+            fetchTotalActiveBalance()
```

Comment:
We forgot to guard this one with a `if (!cancelled)`.

However, what we probably want to do anyway is use a `.finally()` on the promise as we do this on both the resolve and reject cases.

Reply:
Yes!! Very good point